### PR TITLE
Add getTicks2 to support condition codes

### PR DESCRIPTION
**Pull request by joel23888**
Addresses #117

### CODE REVIEW

**Comment by joel23888** [comment]
@eddelbuettel do you have any advice on what I need to do here, if anything?

**Comment by eddelbuettel** [comment]
Sure. Look at the Travis log, line 723 has 'Rblpapi_types.h not found' so somehow you messed things up.  As such the PR is no good.

```
Setting up compilation for a linux 64-bit system
** libs
g++ -std=c++0x -I/usr/share/R/include -DNDEBUG -I../inst/include/ -I.  -I"/usr/lib/R/site-library/Rcpp/include" -I"/usr/lib/R/site-library/BH/include"   -fpic  -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Wformat-security -Werror=format-security -D_FORTIFY_SOURCE=2 -g -c RcppExports.cpp -o RcppExports.o
RcppExports.cpp:4:27: fatal error: Rblpapi_types.h: No such file or directory
compilation terminated.
make: *** [RcppExports.o] Error 1
```

Now, in my source directory I see only `rblpapi_types.h`.  Did you by chance work on a Mac where case may have gotten mixed?

```bash
edd@max:~/git/rblpapi(master)$ grep rblpapi_types.h src/*
src/blpapi_utils.h:#include <rblpapi_types.h>
edd@max:~/git/rblpapi(master)$ 
```

File: src/RcppExports.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // This file was generated by Rcpp::compileAttributes
 // Generator token: 10BE3573-1514-4C36-9D1C-5A225CD40393
 
+#include "Rblpapi_types.h"
```
Comment (eddelbuettel) [inline src/RcppExports.cpp]: That is odd.   Rcpp sometimes creates `NAMEOFPACKAGE_types.h` but it hasn't done that for me here.

And we don't have.  But we now have `rblpapi_types.h`.   I wonder if it got confused.

Comment (eddelbuettel) [inline src/RcppExports.cpp]: What version of Rcpp do you have installed?

When I add a quick testfile `src/foo.cpp` and re-run `compileAttributes.h` I do *not* get an additional export for `Rblpapi_types.h`.  Something is different at your side, and you need to look into that.

**Comment by joel23888** [comment]
Thank you for getting back to me. The machine I was working on is win64. I do not have access to it right now so I will have to follow up on this tomorrow.

**Comment by joel23888** [comment]
I have version Rcpp 0.12.3 installed. My source directory also only has `rblpapi_types.h`. When I run Rcpp::compileAttributes it adds `#include "Rblpapi_types.h"` to RcppExports.cpp.

**Comment by joel23888** [comment]
FWIW (and I am not proposing to do this) but changing the package name in DESCRIPTION to rblpapi makes Rcpp::compileAttributes add #include "rblpapi_types.h" to RcppExports.cpp instead. Not sure if this is obvious or helps at all.

**Comment by eddelbuettel** [comment]
It does smell like an issue in Rcpp.  I lobbed a question to my co-conspirators but have not yet heard back.

There is a snowball's chance in hell for renaming the package (cf CRAN Repo Policy) but we can rename the file.  I can do that in a bit.  If I do so, can you merge the change into your branch and update?

**Comment by eddelbuettel** [comment]
And once your PR builds we can talk about integrating. I am not a fan of doubling and tripling existing functions -- so instead of adding `getTicks2()` I'd rather see us work on extending `getTicks()` to supply the functionality needed given proper parameters.

**Comment by joel23888** [comment]
Yes, I agree.

**Comment by eddelbuettel** [comment]
@joel23888 If you pull now, and the merge from master into your branch, things should build as we just got PR #145 in.

**Comment by joel23888** [comment]
It built correctly but would you mind if I redo this with the correct username? I forgot to update my username for this project as it differs from my global settings.

**Comment by joel23888** [comment]
Closing this and will reopen after tidying up something on my side.

**Pull request closed without merging**